### PR TITLE
feat(tokens): make font-size use px-to-rem()

### PR DESCRIPTION
**Pull request by BenOsodrac**
Issue number: internal

---------

## What is the new behavior?

This PR intends to make sure all font-size related tokens are generated suing the px-to-rem() scss function, so that the fonts correctly scale in runtime to zoom/A11y preferences.
 
- Added new utility to generate the expected css and scss variable using the px-to-rem()
- Added @use on generated scss and css foundations files.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/foundations/ionic.root.scss [hunk 1]
```diff
@@ -12,18 +14,18 @@
...
+  --ionic-font-size-350: px-to-rem(14);
+  --ionic-font-size-400: px-to-rem(16);
+  --ionic-font-size-450: px-to-rem(18);
+  --ionic-font-size-500: px-to-rem(20);
+  --ionic-font-size-550: px-to-rem(22);
+  --ionic-font-size-600: px-to-rem(24);
+  --ionic-font-size-650: px-to-rem(26);
+  --ionic-font-size-700: px-to-rem(28);
+  --ionic-font-size-800: px-to-rem(32);
+  --ionic-font-size-900: px-to-rem(36);
```
Comment (brandyscarney) [inline core/src/foundations/ionic.root.scss:28]: This function needs to be escaped I believe
```suggestion
  --ionic-font-size-275: #{px-to-rem(11)};
  --ionic-font-size-300: #{px-to-rem(12)};
  --ionic-font-size-350: #{px-to-rem(14)};
  --ionic-font-size-400: #{px-to-rem(16)};
  --ionic-font-size-450: #{px-to-rem(18)};
  --ionic-font-size-500: #{px-to-rem(20)};
  --ionic-font-size-550: #{px-to-rem(22)};
  --ionic-font-size-600: #{px-to-rem(24)};
  --ionic-font-size-650: #{px-to-rem(26)};
  --ionic-font-size-700: #{px-to-rem(28)};
  --ionic-font-size-800: #{px-to-rem(32)};
  --ionic-font-size-900: #{px-to-rem(36)};
```

Comment (BenOsodrac) [reply core/src/foundations/ionic.root.scss]: Thats an epic catch, thank you!

Comment (BenOsodrac) [reply core/src/foundations/ionic.root.scss]: But will need to adjust the script. Will commit in a bit!

**Review by brandyscarney — APPROVED** [review]
Code works great 👍

**Pull request merged**